custom/icds_reports/ucr/data_sources/infrastructure_form.json

```diff
@@ -501,21 +501,55 @@
         "column_id": "medicine_kits_available"
       },
       {
-        "filter": {
-          "operator": "eq",
-          "expression": {
-            "datatype": "string",
-            "type": "property_path",
-            "property_path": [
-              "form",
-              "supply_position_available_1",
-              "preschool_kit"
-            ]
+        "expression": {
+          "type": "conditional",
+          "test": {
+            "operator": "eq",
+            "expression": {
+              "datatype": "string",
+              "type": "property_path",
+              "property_path": [
+                "form",
+                "supply_position_available_1",
+                "preschool_kit"
+              ]
+            },
+            "type": "boolean_expression",
+            "property_value": "yes"
           },
-          "type": "boolean_expression",
-          "property_value": "yes"
+          "expression_if_true": {
+            "type": "constant",
+            "constant": 1
+          },
+          "expression_if_false": {
+            "type": "conditional",
+            "test": {
+              "operator": "eq",
+              "expression": {
+                "datatype": "string",
+                "type": "property_path",
+                "property_path": [
+                  "form",
+                  "supply_position_available_1",
+                  "preschool_kit"
+                ]
+              },
+              "type": "boolean_expression",
+              "property_value": "no"
+            },
+            "expression_if_true": {
+              "type": "constant",
+              "constant": 0
```

Comment:
Don't understand this, but curious why the `expression_if_true` value is a constant zero? Looks like the same construct exists below.

Reply:
This is nasty, kind of like nested if's in xpath. If it were xpath it would look like

`if(preschool_kit = 'yes', 1, if(preschool_kit = 'no', 0, null)`

Reply:
I did it this way to keep similar behavior to the boolean type